Setup and mobile version skeleton

IsraArvin opened the pull request:
##   Implementation of Toolbar and Headline Sections 🚀

👋 Hi team!

I'm thrilled to submit my pull request for this project, which focuses on implementing the Toolbar and Headline sections. 🎉 I've ensured that all requirements were met and followed the best practices for HTML and CSS.

### Changes Made

- Created a new feature branch named `Feature/mobile-skeleton`.
- Implemented the Toolbar section according to the provided design.
- Utilized Flexbox for layout and responsiveness in both the Toolbar and Headline sections.
- Styled the elements to match the design, including fonts, colors, images, text, and margins.
- Added button interactions such as hover effects.
- Ensured responsiveness by using responsive values in CSS rules, like percentages instead of pixels.
-  Revise the README file to include the project details and instructions

### Linter and GitHub Flow

- Verified that the linter is set up correctly for this repository.
- Confirmed that there are no linter errors, resulting in green status.
- Made no custom changes to the linter configuration.
- Followed GitHub Flow throughout the development process.
- Created meaningful commit messages and maintained a descriptive Git history.
- Provided a descriptive title for this pull request to clearly indicate the changes inside.
- Included a summary in this pull request, elaborating on the introduced changes.

### Documentation and Branching

- Documented my work professionally, adhering to the project's requirements.
- Chose the appropriate branch name, `Feature/mobile-skeleton`, to reflect the implemented feature.
- Created this pull request from the feature branch into the main branch, ensuring a proper workflow.

I have also incorporated the Figma design templates and ensured that the website looks exactly like the design, including fonts, colors, images, text, and margins.

🔗 [Figma Design Mockup](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=0%3A1&mode=dev)

Please review my changes and provide any feedback or suggestions for improvement. I'm enthusiastic about collaborating further on this project!

Thank you. 😊

IsraArvin committed:
Initialize GitHub Actions workflow for linters

IsraArvin committed:
Exclude node_modules from version control by adding to .gitignore

IsraArvin committed:
Set up webhint and stylelint linters in local environment

IsraArvin committed:
Add 'images' folder to project

IsraArvin committed:
Implement: Add header and hero section with social media links

This commit implements the initial version of the header and hero section for the website. The header includes a logo and hamburger menu icon, while the hero section provides an introduction to IsaMarvin as a software engineer.

Commit Description:
- Implemented header section with logo and hamburger menu icon
- Added hero section with heading, description, and "Let's Connect" prompt
- Included social media links for Twitter, LinkedIn, Medium, GitHub, and AngelList

IsraArvin committed:
Add custom font and reset styles for body and its elements

- Added a custom font import for Poppins from the Google Fonts API.
- Set the font-family of the root element to "Poppins" and fallback to sans-serif.
- Reset the margin, box-sizing, padding, font-size, color, and text-decoration for the body element and all its child elements.

IsraArvin committed:
Implement initial CSS styling for header and hero section

- Implemented the initial CSS styling for the header and hero section of the website.
- Set global box-sizing and margin/padding rules for consistent layout and spacing.
- Defined color variables for various shades and used them throughout the styling.
- Set a background color for the HTML element to create a neutral background.
- Styled the header with flexbox properties for proper alignment, padding, and background color.
- Added a background image to the wrapper element with a cover size and border radius.
- Styled the hero section with flexbox properties, proper margin, font styles, and color.
- Added a hover effect to social media icons in the social section.

IsraArvin committed:
Add README with project details and instructions

IsraArvin committed:
Include up-to-date linter configuration files

mckent05 commented on index.html:
```diff
@@ -0,0 +1,69 @@
+    <title>IsaMarvin | Software Engineer</title>
+  </head>
+  <body>
+    <header class="header">
+      <img src="./images/my-logo.svg" alt="logo" />
+      <img
+        src="./images/hamberger.svg"
+        alt="hamburger"
+      />
+    </header>
```
- Good job wrapping this section in a `header` tag, but to help with it semantics of your page, please wrap your logo and hamburger icon in a `nav` tag 🙏

mckent05 commented on index.html:
```diff
@@ -0,0 +1,69 @@
+        </p>
+        <p class="hero__connect">LET’S CONNECT</p>
+        <div class="social">
+          <a href="#" class="social__link">
+            <img
+              class="social__icon"
+              src="./images/social/twitter.svg"
+              alt="twitter"
+            />
+          </a>
+          <a href="#" class="social__link">
+            <img
+              class="social__icon"
+              src="./images/social/linkedin.svg"
+              alt="linkedin"
+            />
+          </a>
+          <a href="#" class="social__link">
+            <img
+              class="social__icon"
+              src="./images/social/medium.svg"
+              alt="medium"
+            />
+          </a>
+          <a href="#" class="social__link">
+            <img
+              class="social__icon"
+              src="./images/social/github.svg"
+              alt="github"
+            />
+          </a>
+          <a href="#" class="social__link">
+            <img
+              class="social__icon"
+              src="./images/social/vector.svg"
+              alt="angellist"
+            />
+          </a>
+        </div>
+      </div>
```
- You have done well so far, but to improve the semantics of your page, I suggest you wrap this section in a `ul` tag and wrap all `a` tags in a corresponding `li` tag. Please fix this 👍 👍

mckent05 reviewed (changes requested):
Hi @IsaMarvin, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
![almost there](https://user-images.githubusercontent.com/73607512/207265056-d056ea78-77c5-4b66-802e-694a5ed4ecf3.gif)

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- You did well making your portfolio responsive across all mobile screens 👏 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

IsraArvin committed:
Commit message: Enhance page semantics by updating HTML structure

Changes:
- Wrapped the logo and hamburger icon elements in a <nav> tag for improved semantics.
- Updated the section structure by wrapping it in a <ul> tag and each <a> tag in a corresponding <li> tag to enhance page semantics.

KanzaTahreem commented on index.html:
```diff
@@ -0,0 +1,73 @@
+  <body>
+    <header class="header">
+      <nav>
+        <img src="./images/my-logo.svg" alt="logo" />
```
- [OPTIONAL] I think it would be better to use `a` tag for logo 🔗 This is for three reasons: convention, semantics, and user experience(UX). [Check this link](https://ux.stackexchange.com/questions/81727/why-is-it-standard-for-a-website-logo-to-navigate-to-the-home-page) for more info ✨

KanzaTahreem commented on README.md:
```diff
@@ -1 +1,152 @@
+
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](../../issues/).
```
- [OPTIONAL] The issues link here is redirecting to a **_page not found_** error ⚠ This can be fixed by using the absolute path for the issues page instead  of the relative page. You can consider 👇

  `[issues page](https://github.com/IsaMarvin/Portfolio_Microverse/issues)`

KanzaTahreem reviewed (approved):
# Approved 🙌

Hi @IsaMarvin 

Great work implementing the suggested changes 💪 **_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**

<img src="https://media.tenor.com/xDHCe07zrocAAAAC/congrats-minions.gif"/>

## Highlights 🎯
There are some excellent points that you have made perfectly. Keep up the great work! 👏

-  ✅ Styling is looking great, perfect!
-  ✅ Great work using the right GitHub Flow 
-  ✅ Your README looks good, well done!
-  ✅ Linters are working properly, great job! 

## Optional suggestions ❕
N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
